.github/workflows/ci.yml

```diff
@@ -466,103 +473,126 @@ jobs:
+              -p mc-fog-sql-recovery-db \
+              -p mc-fog-test-client \
+              -p mc-fog-view-server \
+              -p mc-ledger-distribution \
+              -p mc-mobilecoind \
+              -p mc-mobilecoind-dev-faucet \
+              -p mc-util-generate-sample-ledger \
+              -p mc-util-grpc-admin-tool \
+              -p mc-util-keyfile \
+              -p mc-util-seeded-ed25519-key-gen \
+              --release
+
+          BIN_DIR="$PWD/target/release"
+
+          # Run in temp dir to appease check-dirty-git.
+          mkdir -p /tmp/fog-local-network
+          cd /tmp/fog-local-network
+
+          # Generate sample keys and ledger.
+          FOG_AUTHORITY_ROOT=$("$BIN_DIR/mc-crypto-x509-test-vectors" --type=chain --test-name=ok_rsa_head)
+          "$BIN_DIR/sample-keys" --num 10 --seed=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+          "$BIN_DIR/generate-sample-ledger" --txs 100
+
+          # Generate sample Fog keys.
+          "$BIN_DIR/sample-keys" --num 4 --output-dir fog_keys --fog-report-url 'insecure-fog://localhost:6200' --fog-authority-root $FOG_AUTHORITY_ROOT
+
+          service postgresql start
+      - name: Run local network
+        env:
+          # TEST_DATABASE_URL points at the server, as Fog recovery DB tests
+          # create and drop PG databases.
+          TEST_DATABASE_URL: postgres://localhost
+        run: |
+          BIN_DIR="$PWD/target/release"
+          SCRIPT_DIR="$PWD/tools/fog-local-network"
+          STRATEGIES_DIR="$PWD/mobilecoind/strategies"
+          export MC_CHAIN_ID="local"
+          export MC_LOG=info
+
+          cd /tmp/fog-local-network
+          export LEDGER_BASE="$PWD/ledger"
+
+          # Run local network in background.
+          MC_LOG="info,rustls=warn,hyper=warn,tokio_reactor=warn,mio=warn,want=warn,rusoto_core=error,h2=error,reqwest=error,rocket=error,<unknown>=error" \
+          python3 "$SCRIPT_DIR/fog_local_network.py" --network-type dense5 --skip-build &
+          # Give it time to spin up
+          for PORT in 3200 3201 3202 3203 3204 4444; do
+            for i in $(seq 0 60); do
+              if ss -l | grep -q ":$PORT"; then break; else sleep 1; fi;
+            done
+          done
+
+          # Authorize minters
+          echo "Authorizing minters"
+          python3 "$SCRIPT_DIR/../local-network/authorize-minters.py"
+          sleep 20
+
+          # Mint 1 million token1's to the first 4 accounts
+          echo "Minting"
+          for ACCOUNT_NUM in $(seq 0 4); do
+              "$BIN_DIR/mc-consensus-mint-client" \
+                  "generate-and-submit-mint-tx" \
+                  --node insecure-mc://localhost:3200/ \
+                  --signing-key "$BIN_DIR/mc-local-network/minting-keys/minter1" \
+                  --recipient $(cat keys/account_keys_${ACCOUNT_NUM}.b58pub) \
+                  --token-id 1 \
+                  --amount 1000000
+          done
+
+          sleep 60
```

Comment:
Consider replacing this sleep with a poll loop

Reply:
I'm not sure how to do that easily -- so like, i'm kind of waiting for the network to be quiet and these transactions to have all "landed" so that when we go to burn, the balance is not 0

Reply:
maybe we can poll on like, something that tells us the block index, and if e.g. 5 seconds go by without it changing then we assume it's done?

Reply:
that tool doesn't really exist atm

Reply:
i guess i could make it, but i'd rather make a ticket to try to do that later?

Reply:
if that tool exists i would replace the sleep 20 above with it also

Reply:
i opened an issue for this